terraformutils/utils.go

```diff
@@ -65,40 +65,86 @@ func PrintTfState(resources []Resource) ([]byte, error) {
 	return buf.Bytes(), err
 }
 
-func RefreshResources(resources []Resource, provider *providerwrapper.ProviderWrapper) ([]Resource, error) {
-	refreshedResources := []Resource{}
-	input := make(chan *Resource, 100)
+func RefreshResources(resources []*Resource, provider *providerwrapper.ProviderWrapper, slowProcessingResources [][]*Resource) ([]*Resource, error) {
+	refreshedResources := []*Resource{}
+	input := make(chan *Resource, len(resources))
 	var wg sync.WaitGroup
 	poolSize := 15
-	if slowProcessingRequired(resources) {
-		poolSize = 1
+	for i := range resources {
+		wg.Add(1)
+		input <- resources[i]
 	}
+	close(input)
```

Comment:
why it's here?

Reply:
@sergeylanzman Because we're done adding data to the channel at that point.